src/library_browser.js

```diff
@@ -1164,6 +1164,12 @@ var LibraryBrowser = {
       GL.newRenderingFrameStarted();
 #endif
 
+      // If the current GL context is an OffscreenCanvas, but it was initialized with implicit swap mode, perform the swap
+      // in behalf of the user.
+      if (typeof GL !== 'undefined' && GL.currentContext && !GL.currentContext.attributes.explicitSwapControl && GL.currentContext.GLctx.commit) {
```

Comment:
Consulting with Morris Tseng who wrote our offscreen canvas support, he says that currently if an onscreen context has ever been created with the canvas, it is no longer possible to transfer the control of the canvas to offscreen, and also, when one transfers the control of the canvas to offscreen, it permanently taints the canvas and it is no longer possible to create onscreen contexts. The only way to create an onscreen context in this situation would be to delete the canvas DOM object and recreate it. I think it is better to have this .commit support in the main thread as well.

As a sidenote, I don't think the OffscreenCanvas API design is very good with this respect, since this is a case where initialization has been thought of, but deinitialization hasn't.